### DISCUSSION TITLE
Restoring single folder / file from Borg Backup

### DISCUSSION THREAD

**Discussion by elephantastyczny** [nextcloud/all-in-one#3330] [Questions] [answered]
Hi All,

I'm trying to recover a folder / file that LDAP user had in Documents folder on Nexcloud AIO.
Since I've purged all data from Nextcloud instance, my only hope is to recover it from backup that Borg is doing daily.

Any hints how I can connect to AIO BorgBackup repository in order to browse an archive?

**Comment by rlew-is** [comment, marked as answer]
So I just tried this with my own backup and you can restore individual files from user directories with the following steps:
Assuming: your archive is in a "borg/" directory and it's next to a temp/ directory. You will need the backup passphrase.
1. Run borg on the nextcloud-aio host and mount borg/ into temp/ `borg mount borg/ temp/`
2. Enter your passphrase.
3. cd into `temp/[date]_[time]-nextcloud-aio/nextcloud_aio_volumes/nextcloud_aio_nextcloud_data/[username]/files/Documents`
4. Copy out the file however you like. `cp [file] /other/dir/`

**Reply by elephantastyczny** [reply]
Brilliant, thanks @rlew-is!

**Reply by mstrauss2018** [reply]
I tried this.
when I ran borg mount it asked for my passphrase 3 times. Each time I pasted it in.
I wen to the AIO interface and copied the encryption password for backups listed under the Backup & Restore section.
every file I ran the cat command on only listed the word Putty many times.
Did it ask for the passphrase 3 times because it was the wrong one? If so, then where do I get my passphrase?
I've been trying for over 1 week to try to restore 1 file and this is the closest I've gotten so far.

**Reply by elephantastyczny** [reply]
Yes, it means that you've entered your password wrong 3 times

You can find borg secret in /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/*backupsecret

![image](https://github.com/nextcloud/all-in-one/assets/120596179/43a832bc-9537-4630-a784-3f04caaa8384)
